operations/helm/charts/mimir-distributed/templates/_helpers.tpl

```diff
@@ -742,3 +782,57 @@ kubectl image reference
 {{- define "mimir.kubectlImage" -}}
 {{ .Values.kubectlImage.repository }}:{{ .Values.kubectlImage.tag }}
 {{- end -}}
+
+{{/*
+TLS configuration template for internal Mimir components
+*/}}
+{{- define "mimir.serverTlsConfig" -}}
+cert_file: {{ .Values.tls.mimir.server.cert_path }}
+key_file: {{ .Values.tls.mimir.server.key_path }}
+client_ca_file: {{ .Values.tls.mimir.server.ca_path }}
+client_auth_type: {{ .Values.tls.mimir.server.client_auth_type }}
+{{- end }}
+
+{{- define "mimir.clientTlsConfig" -}}
+tls_enabled: {{ .Values.tls.mimir.enabled }}
+{{- if .Values.tls.mimir.enabled }}
+tls_cert_path: {{ .Values.tls.mimir.client.cert_path }}
+tls_key_path: {{ .Values.tls.mimir.client.key_path }}
+tls_ca_path: {{ .Values.tls.mimir.client.ca_path }}
+tls_server_name: {{ .Values.tls.mimir.client.server_name }}
+tls_insecure_skip_verify: {{ .Values.tls.mimir.client.insecure_skip_verify }}
+tls_cipher_suites: {{ .Values.tls.mimir.client.cipher_suites }}
+tls_min_version: {{ .Values.tls.mimir.client.min_version }}
+{{- end }}
+{{- end }}
+
+{{- define "memcached.clientTlsConfig" -}}
+tls_enabled: {{ .Values.tls.memcached.enabled }}
+{{- if .Values.tls.memcached.enabled }}
+tls_cert_path: {{ .Values.tls.memcached.client.cert_path }}
+tls_key_path: {{ .Values.tls.memcached.client.key_path }}
+tls_ca_path: {{ .Values.tls.memcached.client.ca_path }}
+tls_server_name: {{ .Values.tls.memcached.client.server_name }}
+tls_insecure_skip_verify: {{ .Values.tls.memcached.client.insecure_skip_verify }}
+tls_cipher_suites: {{ .Values.tls.memcached.client.cipher_suits }}
+tls_min_version: {{ .Values.tls.memcached.client.min_version }}
+{{- end }}
+{{- end }}
+
+{{- define "memcachedExporter.serverTlsConfig" -}}
+{{- if .Values.tls.memcached.enabled }}
+tls_server_config:
+    cert_file: {{ .Values.tls.memcached.server.cert_path }}
+    key_file: {{ .Values.tls.memcached.server.key_path }}
+{{- end }}
+{{- end }}
+
+{{/*
+TLS protocol template for Mimir components
+*/}}
+{{ define "mimir.internalProtocol" -}}
+http{{- if .Values.tls.mimir.enabled -}}s{{- end -}}
+{{- end }}
+{{ define "memcached.internalProtocol" -}}
```

Comment:
let's have this prefixed with `mimir.` I don't want to conflict with some obscure memcached chart someone decided to use